core/domain/caching_services.py

```diff
@@ -0,0 +1,238 @@
+# coding: utf-8
+#
+# Copyright 2014 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Service functions to set and retrieve data from the memory cache."""
+
+from __future__ import absolute_import  # pylint: disable=import-only-modules
+from __future__ import unicode_literals  # pylint: disable=import-only-modules
+
+from core.domain import collection_domain
+from core.domain import exp_domain
+from core.domain import skill_domain
+from core.domain import story_domain
+from core.domain import topic_domain
+from core.platform import models
+import python_utils
+
+memory_cache_services = models.Registry.import_cache_services()
+
+# NOTE: Namespaces cannot contain ':'.
+# Sub-namespace is defined as the version number of the Exploration. The
+# namespace handles Exploration objects and returns an Exploration objects in
+# the format of a dictionary with string to Exploration id-value pairs.
+CACHE_NAMESPACE_EXPLORATION = 'exploration'
+# Sub-namespace is defined as the version number of the Collection. The
+# namespace handles Collection objects and returns a Collection objects in
+# the format of a dictionary with string to Collection id-value pairs.
+CACHE_NAMESPACE_COLLECTION = 'collection'
+# Sub-namespace is defined as the version number of the Skill. The namespace
+# handles Skill objects and returns Skill objects in
+# the format of a dictionary with string to Skill id-value pairs.
+CACHE_NAMESPACE_SKILL = 'skill'
+# Sub-namespace is defined as the version number of the Story. The namespace
+# handles Story object and returns Story objects in
+# the format of a dictionary with string to Story id-value pairs.
+CACHE_NAMESPACE_STORY = 'story'
+# Sub-namespace is defined as the version number of the Topic. The namespace
+# handles Topic objects and returns Topic objects in
+# the format of a dictionary with string to Topic id-value pairs.
+CACHE_NAMESPACE_TOPIC = 'topic'
+# Sub-namespace is not defined. The namespace handles a ConfigPropertyModel
+# value and returns a ConfigPropertyModel value(returned from the value
```

Comment:
Anytime you type "(", have a single space before it, unless you're doing a function call.